### PR TITLE
Add camera zoom plugin

### PR DESCRIPTION
**Pull request by srmainwaring**
Add a 3 axis gimbal and system plugin that attaches to a camera sensor and allows the camera to be zoomed.

## Details

- Add 3 axis version of the small gimbal.
- Add example world with mount and objects of interest.
- Add system plugin that subscribes to a zoom command topic (`msgs::Double`) and allows an existing camera sensor to be zoomed.

The plugin SDF has a single parameter setting the maximum zoom factor.

```xml
 <sensor name="camera" type="camera">
   ...
   <plugin filename="CameraZoomPlugin"
       name="CameraZoomPlugin">
     <max_zoom>15.0</max_zoom>
   </plugin>
</sensor>
```

## Testing

https://github.com/ArduPilot/ardupilot_gazebo/assets/24916364/57dd7fb5-3db1-4ab1-b8ed-ea516f9caa96

The gimbal is position controlled and responds to roll, pitch and yaw commands.

```bash
gz topic -t "/gimbal/cmd_yaw" -m gz.msgs.Double -p "data: -1"
```

The camera zoom is similarly commanded.

```bash
gz topic -t "/model/gimbal/sensor/camera/zoom/cmd_zoom" -m gz.msgs.Double -p "data: 2"
```

## Tasks

- [x] `#include <gz/sim/Sensor.hh>` is not available in the released version of `gz-sim7` on Ubuntu.
- [x] Resolve segmentation fault on exit.

### CODE REVIEW

File: src/CameraZoomPlugin.cc [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+namespace gz {
+namespace sim {
+inline namespace GZ_SIM_VERSION_NAMESPACE {
+namespace systems {
+
+//////////////////////////////////////////////////
+class CameraZoomPlugin::Impl
+{
+  /// \brief Handle a zoom command.
+  public: void OnZoom(const msgs::Double &_msg);
```
Comment (Ryanf55) [inline src/CameraZoomPlugin.cc:65]: Have you considered specifying zoom as a focal length? It's more deterministic.

Comment (srmainwaring) [reply src/CameraZoomPlugin.cc:65]: Yes, could do. This is just a basic first pass to get the internal mechanics of the plugin working and a factor applied to HFOV was the method used in the PX4-SITL_Gazebo plugin for Gazebo Classic so I took my steer from that.

**Comment by rmackay9** [comment]
It's really great to see this, thanks so much!

**Comment by Ryanf55** [comment]
If you are still interested to get this merged, let me know! I think it's a great feature addition.

**Comment by srmainwaring** [comment]
> If you are still interested to get this merged, let me know! I think it's a great feature addition.

Thanks @Ryanf55, let me re-test on Harmonic. IIRC there was an issue in Gazebo Garden that could cause crashes and I need to remind my self what it was and whether it has been fixed.

**Comment by srmainwaring** [comment]
> If you are still interested to get this merged, let me know! I think it's a great feature addition.

Hi @Ryanf55 - tested on Harmonic and appears to now be working fine. I can not longer see a segmentation fault on exit, so I assume the issues (which was connected to the physics libraries unloading IIRC) has been resolved upstream.

**Review by Ryanf55 — APPROVED** [review]

**Pull request merged**